src/components/pages/CompanySubscriptions/index.tsx

```diff
@@ -0,0 +1,320 @@
+  const [appId, setAppId] = useState<string>('')
+  const [subscriptionId, setSubscriptionId] = useState<string>('')
+  const [unsubscribeMutation] = useUnsubscribeAppMutation()
+  const [enableErrorMessage, setEnableErrorMessage] = useState<boolean>(false)
+
+  const setView = (e: React.MouseEvent<HTMLInputElement>) => {
+    const viewValue = e.currentTarget.value
+    setFilterStatus(viewValue)
+    setGroup(viewValue)
+    setRefresh(Date.now())
+  }
+
+  const filterView = [
+    {
+      buttonText: t('content.companySubscriptions.filter.requested'),
+      buttonValue: CompanySubscriptionFilterType.REQUESTED,
+      onButtonClick: setView,
+    },
+    {
+      buttonText: t('content.companySubscriptions.filter.active'),
+      buttonValue: CompanySubscriptionFilterType.ACTIVE,
+      onButtonClick: setView,
+    },
+    {
+      buttonText: t('content.companySubscriptions.filter.showAll'),
+      buttonValue: CompanySubscriptionFilterType.SHOW_ALL,
+      onButtonClick: setView,
+    },
+  ]
+
+  useEffect(() => {
+    if (onValidate(searchExpr)) {
+      setFetchHookArgs({
+        statusFilter: filterStatus,
+        expr: searchExpr,
+      })
+    }
+  }, [filterStatus, searchExpr])
+
+  const onValidate = (expr: string) => {
+    const validateExpr = isName(expr)
+    if (validateExpr) dispatch(setSearchInput({ open: true, text: expr }))
+    return validateExpr
+  }
+
+  const onUnsubscribeSubmit = async () => {
+    setLoading(true)
+    await unsubscribeMutation(subscriptionId)
+      .unwrap()
+      .then(() => {
+        success(t('content.organization.unsubscribe.unsubscribeSuccess'))
+        setLoading(false)
+        setShowUnsubscribeOverlay(false)
+        setEnableErrorMessage(false)
+        setRefresh(Date.now())
+      })
+      .catch(() => {
+        setLoading(false)
+        setEnableErrorMessage(true)
+      })
+  }
+
+  const renderStatusButton = (status: string) => {
+    if (status === SubscriptionStatus.ACTIVE)
+      return (
+        <Button
+          variant="text"
+          startIcon={<CheckCircleOutlineIcon />}
+          sx={{ color: '#B3CB2D', pointerEvents: 'none' }}
+          size="small"
+        >
+          {t('content.companySubscriptions.subscribed')}
+        </Button>
+      )
+    else if (status === SubscriptionStatus.PENDING)
+      return (
+        <Button
+          variant="outlined"
+          sx={{
+            color: '#FFA600',
+            borderColor: '#FFA600',
+            pointerEvents: 'none',
+            ml: 1,
+          }}
+          size="small"
+        >
+          {t('content.companySubscriptions.requested')}
+        </Button>
+      )
+    else
+      return (
+        <Button
+          variant="text"
+          startIcon={<UnpublishedIcon />}
+          sx={{ color: '#D91E18', pointerEvents: 'none' }}
+          size="small"
+        >
+          {t('content.companySubscriptions.declined')}
+        </Button>
+      )
+  }
+
+  const columns = [
```

Comment:
to reduce the burden on here, this can be moved as a new file just like registrationTableColumns.tsx